api/test/unit/controllers/test_search_controller.py

```diff
@@ -759,3 +772,59 @@ def test_post_process_results_recurses_as_needed(
     }
 
     assert wrapped_post_process_results.call_count == 2
+
+
+DEFAULT_RANK_QUERY = [
+    Q("rank_feature", field=field, boost=10000)
+    for field in ["standardized_popularity", "authority_boost"]
+]
+
+
+@pytest.mark.parametrize(
+    ("data", "expected_query_filter", "expected_should"),
+    [
+        pytest.param(
+            {"tag": "art"},
+            [{"term": {"tags.name.keyword": "art"}}],
+            DEFAULT_RANK_QUERY,
+            id="filter_by_tag",
+        ),
+        pytest.param(
+            {"tag": "art, photography"},
+            [{"term": {"tags.name.keyword": "art, photography"}}],
+            DEFAULT_RANK_QUERY,
+            id="filter_by_tag_treats_punctuation_as_part_of_tag",
+        ),
+        pytest.param(
+            {"source": "flickr"},
+            [{"term": {"source.keyword": "flickr"}}],
+            [],
+            id="filter_by_source",
+        ),
+        pytest.param(
+            {"source": "flickr", "creator": "nasa"},
+            [
+                {"term": {"source.keyword": "flickr"}},
+                {"term": {"creator.keyword": "nasa"}},
+            ],
+            [],
+            id="filter_by_creator",
+        ),
+    ],
+)
+@mock.patch("api.controllers.search_controller.Search", wraps=Search)
```

Comment:
I would recommend _against_ mocking `Search`. I did it before in [`test_media_report` but replaced it with pook matchers recently](https://github.com/WordPress/openverse/commit/503ac82ad432df120497674a77fb8d72066c6c7f#diff-59d0ff474f161029e004a8e550ee72d2cb61875dd1650cd079d0079f3e64c852) when it turned out it was actually necessary to check the real calls to Elasticsearch. In particular, I think it was the client API using different parameters, but the ES request remaining the same, which revealed to me that we were more or less testing the implementation details of the method rather than the actual functionality to Elasticsearch.

Not a blocker or strong suggestion, just sharing an observation from some recent work I did with testing against ES.